contrib_bots/lib/github.py

```diff
@@ -0,0 +1,63 @@
+# The purpose of this file is to handle all requests
+# to the Github API, which will make sure that all
+# requests are to the same account, and that all requests
+# authenticated correctly and securely
+from __future__ import absolute_import
+from future import standard_library
+standard_library.install_aliases()
+import json
+import os
+import requests
+import six.moves.configparser
+import urllib.request, urllib.error, urllib.parse
+
+class github(object):
+    '''
+    Before use, check ~/.github/github.conf, and make sure there are
+    github_repo   (The name of the repo to post to)
+    github_repo_owner   (The owner of the repo to post to)
+    github_username   (The username of the github bot)
+    github_token    (The personal access token for the github bot)
+    '''
+    
+    CONFIG_FILE = '~/.github/github.conf'
+    BASE_URL = ''
+    SESSION = requests.session()
```

Comment:
I think SESSION should be an instance variable, not a class-wide variable, since, in theory, somebody using your library may have multiple sessions open.

So you want something more like

```
def __init__(self):
    self.session = requests.session()
```